src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1467,11 +1469,18 @@ protected void ResetEngineInternal()
 
             EngineController.MessageLogged += LogMessage;
             EngineController.TraceReconcliationComplete += EngineController_TraceReconcliationComplete;
+            EngineController.RequestCustomNodeRegistration += EngineController_RequestCustomNodeRegistration;
 
             foreach (var def in CustomNodeManager.LoadedDefinitions)
                 RegisterCustomNodeDefinitionWithEngine(def);
         }
 
+        private void EngineController_RequestCustomNodeRegistration(object sender, EventArgs e)
+        {
+            foreach (var def in CustomNodeManager.LoadedDefinitions)
```

Comment:
I'm curious about the timing of this - as I think @reddyashish mentioned - what if the package to be loaded contains custom nodes, would those exist in the `LoadedDefinitions` list before this call is made?

Reply:
That seems with investigating. Taking a look at the interaction during the `PackgeLoaded` event, fired on the first stage of `LoadPackages`, there are some steps related to custom nodes that I did not fully capture. I'll take a closer look and see if I can find where this happens and add it to the internal documentation.